### PR TITLE
HDDS-7251. Replace Log4j 1.x with Reload4j

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

Log4j 1.x is EOL.  This PR proposes to replace it with Reload4j, which is a drop-in replacement with security updates.

https://issues.apache.org/jira/browse/HDDS-7251

## How was this patch tested?

Built locally.  Only Log4j 2 present in distribution:

```
$ ls -1 hadoop-ozone/dist/target/ozone-1.3.0-SNAPSHOT/share/ozone/lib/*log4j*
hadoop-ozone/dist/target/ozone-1.3.0-SNAPSHOT/share/ozone/lib/log4j-api-2.17.1.jar
hadoop-ozone/dist/target/ozone-1.3.0-SNAPSHOT/share/ozone/lib/log4j-core-2.17.1.jar
```

Verified no Log4j or Reload4j classes are present in shaded FS jars.

Full CI:
https://github.com/adoroszlai/hadoop-ozone/actions/runs/3103286435

### CODE REVIEW

File: hadoop-ozone/s3gateway/pom.xml [hunk 1]
```diff
@@ -147,48 +147,6 @@
       <groupId>org.apache.ozone</groupId>
       <artifactId>hdds-hadoop-dependency-test</artifactId>
       <scope>test</scope>
-      <exclusions>
```
Comment (jojochuang) [inline hadoop-ozone/s3gateway/pom.xml:150]: is this a related change?

**Review by jojochuang — COMMENTED** [review]
Looks fine by me.

It could be made simpler by simply removing direct dependency on log4j:log4j, and update slf4j-log4j12 version to 1.7.36. Starting from slf4j-log4j12 1.7.35, it automatically redirects to slf4j-reload4j and will load the ch.qos.reload4j:reload4j indirectly.

https://www.slf4j.org/manual.html#projectDep

File: hadoop-ozone/s3gateway/pom.xml (hunk 1 above)
Comment (adoroszlai) [reply hadoop-ozone/s3gateway/pom.xml:150]: Yes, it's related, but not strictly required.  I was trying to eliminate `log4j` coming as transitive dependency via various Hadoop test jars e.g. `hadoop-hdfs`.  I replaced dependence on `hadoop-hdfs` with `hdds-hadoop-dependency-test`.  Since the latter already excludes transitive dependencies, these local exclusions seemed unnecessary.

**Review by jojochuang — APPROVED** [review]
+1 LGTM

**Comment by jojochuang** [comment]
Merged. Thanks @adoroszlai

**Comment by adoroszlai** [comment]
Thanks @jojochuang for reviewing and merging this.

File: pom.xml [hunk 2]
```diff
@@ -150,10 +150,11 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xs
     <httpcore.version>4.4.13</httpcore.version>
 
     <!-- SLF4J/LOG4J version -->
-    <slf4j.version>1.7.30</slf4j.version>
+    <slf4j.version>1.7.36</slf4j.version>
     <log4j.version>1.2.17</log4j.version>
```
Comment (smengcl) [inline pom.xml:154]: Thanks @adoroszlai for the patch.

Looks like we can remove the `log4j.version` ? now that it is no longer being used.
